src/Agent.Worker/DiagnosticLogManager.cs

```diff
@@ -198,6 +208,54 @@ public sealed class DiagnosticLogManager : AgentService, IDiagnosticLogManager
             executionContext.Debug("Diagnostic file upload complete.");
         }
 
+        // Dumping Agent Azure VM extension logs to the support files folder.
+        private void DumpAgentExtensionLogs(IExecutionContext executionContext, string supportFilesFolder, DateTime jobStartTimeUtc)
+        {
+            executionContext.Debug("Starting dumping agent extension logs.");
+
+            string pathToLogs = null;
+            string archiveName = null;
+            string timestamp = jobStartTimeUtc.ToString("yyyyMMdd-HHmmss");
+
+            if (PlatformUtil.RunningOnWindows)
+            {
+                string pathToVersions = "C:\\WindowsAzure\\Logs\\Plugins\\Microsoft.VisualStudio.Services.TeamServicesAgent";
+                if (Directory.Exists(pathToVersions))
+                {
+                    string[] subDirs = Directory.GetDirectories(pathToVersions).Select(dir => dir.Split("\\").Last()).ToArray();
+                    Version[] versions = subDirs.Select(dir => new Version(dir)).ToArray();
+                    Version maxVersion = versions.Max();
+                    pathToLogs = Path.Combine(pathToVersions, maxVersion.ToString());
+                }
+                archiveName = $"AgentWindowsExtensionLogs-{timestamp}-utc.zip";
+            }
+
+            if (PlatformUtil.RunningOnLinux)
+            {
+                pathToLogs = "/var/log/azure/Microsoft.VisualStudio.Services.TeamServicesAgentLinux";
+                archiveName = $"AgentLinuxExtensionLogs-{timestamp}-utc.zip";
+            }
+
+            if (Directory.Exists(pathToLogs))
+            {
+                executionContext.Debug($"Path to agent extension logs: {pathToLogs}");
+
+                string archivePath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Diag), archiveName);
+                executionContext.Debug($"Archiving agent extension logs to: {archivePath}");
+                ZipFile.CreateFromDirectory(pathToLogs, archivePath);
+
+                string copyPath = Path.Combine(supportFilesFolder, archiveName);
+                executionContext.Debug($"Copying archived agent extension logs to: {copyPath}");
+                File.Copy(archivePath, copyPath);
```

Comment:
Should we also remove the archive that was created by the pipeline agent in `archivePath`? Or at least use the `move` method instead of `copy`?

Reply:
I suggest leaving the Agent Azure VM Extension logs in the `_diag` folder by analogy with the EventViewer logs.